src/main/java/by/dragonsurvivalteam/dragonsurvival/registry/DSModifiers.java

```diff
@@ -20,23 +22,23 @@
 import net.neoforged.neoforge.common.NeoForgeMod;
 
 import java.util.List;
+import java.util.Optional;
 import java.util.function.Function;
 
 import static by.dragonsurvivalteam.dragonsurvival.DragonSurvival.MODID;
 
 public class DSModifiers {
 
-    public record ModifierBuilder(ResourceLocation modifier, Holder<Attribute> attribute, Operation operation,
-                                Function<DragonStateHandler, Double> calculator) {
-        private AttributeModifier buildModifier(DragonStateHandler handler) {
-            return new AttributeModifier(modifier, calculator.apply(handler), operation);
+    private record ModifierBuilder(ResourceLocation modifier, Holder<Attribute> attribute, Operation operation,
+                                Function<Player, Double> calculator) {
+        private AttributeModifier buildModifier(Player player) {
+            return new AttributeModifier(modifier, calculator.apply(player), operation);
         }
 
         public void updateModifier(Player player) {
```

Comment:
can we pass the dragonstate handler from the original call (which updates all modifiers) (or have an overloaded method for `updateModifier` where we do that)?

Could avoid a lot of calls to get the data attachment